protocol/proto/options.go

```diff
@@ -24,22 +24,13 @@ import (
 	"github.com/m3db/m3x/pool"
 )
 
-const (
-	// Using 16K which provides much better performance comparing
-	// to lower values like 1k ~ 8k.
-	defaultBufferSize = 16384
-)
-
 // NewBaseOptions creates a new BaseOptions.
 func NewBaseOptions() BaseOptions {
-	return &baseOptions{
-		bufferSize: defaultBufferSize,
-	}
+	return &baseOptions{}
 }
 
 type baseOptions struct {
-	bytesPool  pool.BytesPool
-	bufferSize int
+	bytesPool pool.BytesPool
```

Comment:
Is this the only thing shared between encoder and decoder? if so you can probably just get rid of the base options and merge it into encoder/decoder options.

Reply:
yeah that is on my plan, will do in a separate pr